src/itr_evaluator.ml

```diff
@@ -932,8 +932,8 @@ and evaluate_expr (context : 'a context) (e : expr) : record_item =
     else if CCList.for_all is_false evaluated_ors then
       e_false
     else (
-      let simplified_ors =
-        CCList.filter (fun x -> not (is_false x)) evaluated_ors
+      let _ignored_ors, simplified_ors =
+        CCList.partition (fun x -> is_false x) evaluated_ors
```

Comment:
Discussed offline - these are logically equivalent and IMO `filter` is clearer (rather than having to remember which side of the tuple the predicate-passing elements end up on in `partition`)